### ISSUE TITLE
solo5 memory layout printout is not right

### ISSUE THREAD

**Issue by ricarkol** [nabla-containers/rumprun#16]
```
            |      ___|
  __|  _ \  |  _ \ __ \
\__ \ (   | | (   |  ) |
____/\___/ _|\___/____/
Solo5: Memory map: %lu MB addressable:
Solo5:   reserved @ (0x0 - 0x%lx)
Solo5:       text @ (0x%lx - 0x%lx)
Solo5:     rodata @ (0x%lx - 0x%lx)
Solo5:       data @ (0x%lx - 0x%lx)
Solo5:       heap >= 0x%lx < stack < 0x%lx
rump kernel bare metal bootstrap
```

**Comment by ricarkol** [comment]
This works before this solo5 commit: fd869f4813686ca33c207001fbc4623d5e2a6c63

**Comment by ricarkol** [comment]
`vnsprintf` in `bindings/ee_printf.c` is not doing the right thing (still don't know why):

```
int ee_vprintf(const char *fmt, va_list args)
{
...
    n = vsnprintf(buf, PRINTF_BUF_LEN, fmt, args); // the resulting buf is not right
 ...
}
```

```
Breakpoint 3, ee_vprintf (fmt=fmt@entry=0x3254f8 "Solo5: Memory map: %llu MB addressable:\n", args=args@entry=0x1fffff70) at ee_printf.c:706
(gdb) p buf
$2 = "Solo5: Memory map: %lu MB addressable:\n", '\000' <repeats 280 times>

(gdb) p *(int *)(((char *)args[0].reg_save_area)+args[0].gp_offset)
$7 = 512
(gdb) p fmt
$8 = 0x3254f8 "Solo5: Memory map: %llu MB addressable:\n"
```
Both the format string `fmt` and `args` are correct, but `buf` (returned by `vsnprintf`) is not. `buf` should be `Solo5: Memory map: 512 MB addressable:\n`.

**Comment by ricarkol** [comment]
This is a solo5 issue: can reproduce it with solo5 tests (no rumprun involved). Will close this after opening an issue in solo5/solo5.

**Comment by ricarkol** [comment]
fixed with #18
